src/cmd/mk/archive.c

```diff
@@ -138,25 +140,25 @@ atimes(char *ar)
 		return;
 	}
 	while(readn(fd, (char *)&h, sizeof(h)) == sizeof(h)){
-		t = atol(h.date);
+		t = atol(h.p.date);
 		if(t == 0)	/* as it sometimes happens; thanks ken */
 			t = 1;
 		namelen = 0;
-		if(memcmp(h.name, "#1/", 3) == 0){	/* BSD */
-			namelen = atoi(h.name+3);
+		if(memcmp(h.p.name, "#1/", 3) == 0){	/* BSD */
+			namelen = atoi(h.p.name + 3);
 			if(namelen >= sizeof name){
 				namelen = 0;
 				goto skip;
 			}
 			if(readn(fd, name, namelen) != namelen)
 				break;
 			name[namelen] = 0;
-		}else if(memcmp(h.name, "// ", 2) == 0){ /* GNU */
+		}else if(memcmp(h.p.name, "// ", 3) == 0){ /* GNU */
 			/* date, uid, gid, mode all ' ' */
-			for(i=2; i<16+12+6+6+8; i++)
-				if(h.name[i] != ' ')
+			for(i=2; i < SARNAME+12+6+6+8; i++)
+				if(h.h[i] != ' ')
```

Comment:
The crux of this change seems to be this loop, but the fix seems oddly disproportionate to the offense committed; in particular, something like the following feels a little easier to negotiate:

```
static int
allspaces(char* buf, int len)
{
        for(int i=0; i<len; i++)
                if(buf[i] != ' ')
                        return 0;
        return 1;
}

                // and at the call site....
                if(!allspaces(h.name, sizeof(a.name)) ||
                    !allspaces(h.date, sizeof(h.date)) ||
                    !allspaces(h.uid, sizeof(h.uid)) ||
                    !allspaces(h.gid, sizeof(h.gid)) ||
                    !allspaces(h.mode, sizeof(h.mode)))
                        goto skip;
```

Reply:
@dancrossnyc done.